src/js/components/modals/ServiceFormModal.js

```diff
@@ -488,6 +524,39 @@ class ServiceFormModal extends mixin(StoreMixin) {
     );
   }
 
+  getToggleButton() {
+    let classSet = 'modal-form-title-label';
+
+    if (this.shouldDisableForm()) {
+      classSet = `${classSet} blend-out`;
+    }
+
+    return (<ToggleButton
+      className={classSet}
+      checkboxClassName="modal-form-title-toggle-button toggle-button"
+      checked={this.state.jsonMode}
+      onChange={this.handleJSONToggle}>
+      JSON mode
+    </ToggleButton>);
+  }
+
+  getWarningMessage() {
+    let {warningMessage} = this.state;
+    if (!warningMessage) {
+      return null;
+    }
+
+    return (
+      <div>
+        <div className="warning-field">
+          <h4 className="text-align-center flush-top">
```

Comment:
I would also recommend to change the `<h4>` to `<div>`. The margin below looks bad.

![image](https://cloud.githubusercontent.com/assets/883486/17441506/a6e722e4-5b31-11e6-8d66-31cec878afbd.png)